dist/ember-charts.js

```diff
@@ -1583,13 +1598,65 @@ define('ember-charts/components/pie-chart', ['exports', 'module', 'ember', './ch
-    startOffset: _Ember['default'].computed('finishedData', function () {
-      var data = this.get('finishedData');
-      var sum = data.reduce(function (p, d) {
+    // Normally, the pie chart should offset slices so that the largest slice
+    // finishes at 12 o'clock
+    //
+    // However, always setting the largest slice at 12 o'clock can cause significant
+    // difficulty while dealing with label intersections. This problem is exacerbated
+    // when certain configurations of pie charts lead to a high density of
+    // small slice labels at the 6 o'clock or 11:30 positions.
+    //
+    // Therefore, rotate the pie and concentrate all small slices at 8 to 10 o'clock
+    // if there is a high density of small slices inside the pie. This will ensure
+    // that there is plenty of space for labels
+    startOffset: _Ember['default'].computed('finishedData', 'sortKey', 'rotationOffset', function () {
+      var detectDenseSmallSlices = function detectDenseSmallSlices(finishedData) {
+        // This constant determines how many slices to use to calculate the
+        // average small slice percentage. The smaller the constant, the more it
+        // focuses on the smallest slices within the pie.
+        //
+        // Empirically, using a sample size of 2 works very well.
+        var smallSliceSampleSize = 2;
+
+        var sortedData = _.sortBy(finishedData, "percent");
+        var startIndex = 0;
+        var endIndex = Math.min(smallSliceSampleSize, sortedData.length);
+        var largestSlicePercent = _.last(sortedData).percent;
+
+        var averageSmallSlicesPercent = sortedData.slice(startIndex, endIndex).reduce(function (p, d) {
+          return d.percent / (endIndex - startIndex) + p;
+        }, 0);
+
+        // When slices smaller than 2.75 percent are concentrated in any location,
+        // there is a high probability of label intersections.
+        //
+        // However, empirical label intersect evidence has demonstrated that this
+        // threshold must be increased to 5% when there are multiple small slices
+        // from the 5 o'clock to 7 o'clock positions
+        if (averageSmallSlicesPercent <= 2.75) {
+          return true;
+        } else if (averageSmallSlicesPercent <= 5 && 45 <= largestSlicePercent && largestSlicePercent <= 55) {
+          return true;
+        }
+        return false;
+      };
+
+      var finishedData = this.get('finishedData');
+
+      // The sum is not necessarily 100% all of the time because of rounding
+      //
+      // For example, consider finishedData percentages (1.3%, 1.3%, 1.4%, 96%).
+      // They will round to (1%, 1%, 1%, and 96%) when maxDecimalPlace = 0 by
+      // default, which then sums to 99%.
+      var sum = finishedData.reduce(function (p, d) {
         return d.percent + p;
       }, 0);
-      return _.last(data).percent / sum * 2 * Math.PI;
+
+      if (detectDenseSmallSlices(finishedData)) {
```

Comment:
So, if we detect dense small slices, wouldn't we want the `startOffset` to be the sum of rotationOffset and what the offset would've been if we didn't add a rotationOffset? ie: `this.get('rotationOffset') + _.last(finishedData.percent / sum * 2 * Math.PI` ?